burst-junit4/src/test/java/com/squareup/burst/RuleUsingAnnotationsTest.java

```diff
@@ -0,0 +1,68 @@
+package com.squareup.burst;
+
+import java.lang.annotation.Annotation;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runner.RunWith;
+import org.junit.runners.model.Statement;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(BurstJUnit4.class)
+public class RuleUsingAnnotationsTest {
+
+  @Target(METHOD)
+  @Retention(RUNTIME) @interface CustomAnnotation {
+
+  }
+
+  static class RuleWithAnnotation implements TestRule {
```

Comment:
Could be a little simpler if you extend `TestWatcher`, like what [`TestName`](https://github.com/junit-team/junit4/blob/master/src/main/java/org/junit/rules/TestName.java) does.